opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -18,167 +18,21 @@
-                    const BlackoilState&            x,
-                    WellStateFullyImplicitBlackoil& xw);
-    };
-
-
-
-
-    template<class T>
-    SimulatorFullyImplicitBlackoil<T>::SimulatorFullyImplicitBlackoil(const parameter::ParameterGroup& param,
-                                                                   const Grid& grid,
-                                                                   const DerivedGeology& geo,
-                                                                   BlackoilPropsAdInterface& props,
-                                                                   const RockCompressibility* rock_comp_props,
-                                                                   NewtonIterationBlackoilInterface& linsolver,
-                                                                   const double* gravity,
-                                                                   const bool has_disgas,
-                                                                   const bool has_vapoil,
-                                                                   std::shared_ptr<EclipseState> eclipse_state,
-                                                                   BlackoilOutputWriter& output_writer,
-                                                                   const std::vector<double>& threshold_pressures_by_face)
-
-    {
-        pimpl_.reset(new Impl(param, grid, geo, props, rock_comp_props, linsolver, gravity, has_disgas, has_vapoil,
-                              eclipse_state, output_writer, threshold_pressures_by_face));
-    }
-
-
-
-
-
-    template<class T>
-    SimulatorReport SimulatorFullyImplicitBlackoil<T>::run(SimulatorTimer& timer,
-                                                        BlackoilState& state)
-    {
-        return pimpl_->run(timer, state);
-    }
-
-
-    // \TODO: Treat bcs.
-    template<class T>
-    SimulatorFullyImplicitBlackoil<T>::Impl::Impl(const parameter::ParameterGroup& param,
-                                               const Grid& grid,
-                                               const DerivedGeology& geo,
-                                               BlackoilPropsAdInterface& props,
-                                               const RockCompressibility* rock_comp_props,
-                                               NewtonIterationBlackoilInterface& linsolver,
-                                               const double* gravity,
-                                               const bool has_disgas,
-                                               const bool has_vapoil,
-                                               std::shared_ptr<EclipseState> eclipse_state,
-                                               BlackoilOutputWriter& output_writer,
-                                               const std::vector<double>& threshold_pressures_by_face)
+    template<class Implementation>
```

Comment:
I did not want an extra line, but a space like this: `template <class Foo>` instead of `template<class Foo>`. For that matter I do not see a change in the diff?